docs/data.ru.md

```diff
@@ -0,0 +1,137 @@
+|----|-------|------------------|-------------------------|
+| 1  | Анна  | anna@example.com | link_to_avatar_anna.jpg |
+| 2  | Петр  | pert@example.com | link_to_avatar_pert.jpg |
+
+
+Зачем две таблицы вместо одной?
+Чтобы не дублировать данные. Если бы в таблице `Events` были поля `organizerName`, `organizerEmail`, `organizerImage`, 
+то пришлось бы их заполнять каждый раз, когда пользователь создает событие. 
+А если пользователь решит поменять свое имя, то придется обновлять все события,
+которые он создал. Вместо этого, мы храним только идентификатор пользователя, а имя, email и изображение берем из таблицы `Users`.
+Для получения данных о пользователе, который создал событие, 
+нам нужно сделать `JOIN` таблиц `Events` и `Users` по полю `organizerId`.
+
+Пример запроса:
+```sql
+SELECT * FROM Events
+JOIN Users ON Events.organizerId = Users.id
+WHERE Events.id = 1
+```
+
+#### Документоориентированная модель данных
+
+https://www.mongodb.com/docs/manual/core/data-model-design/
+В документоориентированной модели данных, это будет выглядеть следующим образом:
+
+Коллекция `Events`:
+
+Модель mongoose:
+```js
+const UserSchema = new Schema({
+  name: { type: String, required: true },
+  email: { type: String, required: true },
+  image: { type: String, required: true },
+});
+
+const User = mongoose.model('User', UserSchema);
+
+const EventSchema = new Schema({
+  organizerId: { type: Schema.Types.ObjectId, ref: 'User' },
+  tripName: { type: String, required: true },
+  description: { type: String, required: true },
+  isPrivate: { type: Boolean, required: true },
+  startDate: { type: Date, required: true },
+  endDate: { type: Date, required: true },
+});
+
+const Event = mongoose.model('Event', EventSchema);
+
+```
+
+```js
+const events = [
+  {
+    "id": ObjectId("5f8b1d8f0b9c2e1b1c9d4c7a"),
+    "organizerId": ObjectId("5f8b1d8f0b9c2e1b1c9d4c7b"),
+    "tripName": "Велопокатушки",
+    "description": "Мы поедем, мы помчимся",
+    "isPrivate": false,
+    "startDate": "2024-01-01",
+    "endDate": "2024-01-10"
+  },
+  {
+    "id": ObjectId("5f8b1d8f0b9c2e1b1c9d4c7c"),
+    "organizerId": ObjectId("5f8b1d8f0b9c2e1b1c9d4c7d"),
+    "tripName": "Поход в горы",
+    "description": "Поедем в горы",
+    "isPrivate": true,
+    "startDate": "2024-01-11",
+    "endDate": "2024-01-13"
+  }
+]
+```
+Пользователей храним в коллекции `Users`:
+
+```js
+const users = [
+  {
+    "id": ObjectId("5f8b1d8f0b9c2e1b1c9d4c7b"),
+    "name": "Анна",
+    "email": "anna@example.com",
+    "image": "link_to_avatar_anna.jpg"
+  },
+  {
+    "id": ObjectId("5f8b1d8f0b9c2e1b1c9d4c7d"),
+    "name": "Петр",
+    "email": "pert@example.com",
+    "image": "link_to_avatar_pert.jpg"
+  }
+]
+```
+
+Пример запроса события с пользователем, который его создал:
+
+```js
+const event = await Event.findOne({ id: '5f8b1d8f0b9c2e1b1c9d4c7a' }).populate('organizerId');
+```
+
+
+## Бинарные данные
+Изображения, аудио, видео и тд.
+
+// TODO: дописать
```

Comment:
https://www.digitalocean.com/community/tutorials/how-to-upload-a-file-to-object-storage-with-node-js